### PR TITLE
SA-1517 - User Conversion

### PR DESCRIPTION
**Pull request by btstevens**
**What's New:**

Added ability to 'convert' vs 'duplicate' AD profile to local profile. 
Increased Default Temporary password length & complexity.
Updated Windows 10 ADK link to windows 10 2004 release.
Fixed Localadmin check & netbios values in gui.
Utilize SID vs username to allow conversion without cached or healthy secure channel.
Add ability to snapshot windows restore point prior to conversion.

**File Changes Explanation:**

**_Build.ps1_**
- Add build step for uwp_jcadmu.ps1 to uwp_jcadmu.exe

**_Sign.ps1_**
- Add sign step for uwp_jcadmu.exe artifact

**_uwp_jcadmu.ps1_**
- Script that runs on first load of migrated user account if HKCU:\Software\JCADMU exists
- Adds appx_manifest to profile on load

**_ModuleChangelog.md_**
- Module changes added to doc

**_Start-Migration.md_**
- JumpCloudUserName param changed to SelectedUserName

**_uwp_jcadmu.exe_**
- exe packaged uwp_fix_jcadmu.ps1

**_Form.ps1_**
- Bumped version to 1.5.0
- Increase TempPassword to satisfy more domain password policies
- Add 'Convert Profile' Checkbox
- Change netbios name lookup to work when securechannel is in broken state
- Switch local administrator group check to use ADSI
- Add actions when convert checkbox is checked
- Add ConvertProfile to form output object

**_Start-Migration.md_**
- Add ConvertProfile

**_Start-Migration.ps1_**
- Add registry modification functions
- Formatting
- Show SID instead of 'UNKNOWN ACCOUNT' when secure channel is in a broken state, this is to help allow the tool to be used when secure channel is broken.
- Reorder, begin and process block for readability 
- Add active setup keys to run uwpfix executable on first login for migrated accounts
- Create new user account, modify registry, NTFS rights rename and take over keys as required.
- Update Windows 10 ADK link to latest 2004 release

### CODE REVIEW

File: jumpcloud-ADMU/Powershell/Start-Migration.ps1 [hunk 1]
```diff
@@ -5393,58 +5884,35 @@ Function Start-Migration
...
-    $CommandScanStateTemplateCustom = 'cd "{0}amd64\"; .\ScanState.exe "{1}" /config:"{0}config.xml" /i:"{0}miguser.xml" /i:"{0}migapp.xml" /i:"C:\Windows\Temp\custom.xml" /l:"{1}\scan.log" /progress:"{1}\scan_progress.log" /o /ue:"*\*" /ui:"{2}\{3}" /c' # $UserStateMigrationToolVersionPath, $profileStorePath, $netBiosName, $DomainUserName
-    $CommandLoadStateTemplateCustom = 'cd "{0}amd64\"; .\LoadState.exe "{1}" /config:"{0}config.xml" /i:"{0}miguser.xml" /i:"{0}migapp.xml" /i:"C:\Windows\Temp\custom.xml" /l:"{1}\load.log" /progress:"{1}\load_progress.log" /ue:"*\*" /ui:"{2}\{3}" /laC:"{4}" /lae /c /mu:"{2}\{3}:{5}\{6}"' # $UserStateMigrationToolVersionPath, $profileStorePath, $netBiosName, $DomainUserName, $TempPassword, $localComputerName, $JumpCloudUserName
+    write-log -Message("The Selected Migration user is: $SelectedUserName")
+    $SelectedUserSid = CheckUsernameorSID $SelectedUserName
+    if (!$ConvertProfile){
```
Comment (jworkmanjc) [inline jumpcloud-ADMU/Powershell/Start-Migration.ps1:5889]: only set these variables is we are not converting the account

**Comment by jworkmanjc** [comment]
Tested Account Conversion Scenarios:

- DomainAD w/ broken secure channel - converted account ✅
- DomainAD w/ secure channel - converted account ✅
- DomainAD w/ secure channel - USMT migration ✅
- Unbound system from DomainAD - converted account  ✅
- AzureAD Bound - USMT Migration ✅
- AzureAD Bound - converted account ✅
- Unbound system from AzureAD - converted account ✅  (with known exception that user-specific) UWP apps can not be identified for previous user)

**Review by kmoorehead-jc — APPROVED** [review]
Reviewed changes and updated Wiki, went over the bulk of the changes with Joe, did some light testing. All seems good and the updates seem to be really useful.

Approving.

**Review by jworkmanjc — APPROVED** [review]
Approved, just set changelog and documentation.

**Pull request merged**